File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>mpp;
        for(int i=0;i<nums.size();i++) 
        {
            if(mpp.find(target-nums[i])!=mpp.end()) return {i,mpp[target-nums[i]]};
            else  mpp[nums[i]]=i;
        }
        return {};

    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer>mpp=new HashMap<>();
        int arr[]=new int[2];
        for(int i=0;i<nums.length;i++)
        {
            int find=target-nums[i];
            if(mpp.containsKey(find)) {
                arr[0]=i;
                arr[1]=mpp.get(find);
            }
          mpp.put(nums[i],i);
        }
       return arr;


    }
}